evaluate/metrics/assessment.go

```diff
@@ -36,6 +36,8 @@ func RegisterAssessmentKey(key string, multiplier uint64) AssessmentKey {
 var (
 	// AssessmentKeyFilesExecuted holds the successfully executed files.
 	AssessmentKeyFilesExecuted = RegisterAssessmentKey("files-executed", 1)
+	// AssessmentKeyFilesExecutedMaximumReachable holds the maximum theoretically reachable executed files.
+	AssessmentKeyFilesExecutedMaximumReachable = RegisterAssessmentKey("files-executed-maximum-reachable", 0)
```

Comment:
nit: would maybe just call this `files-executed-max` cause we will anyhow know what this is... and once we add things like `coverage-maximum-reachable` these long names really start to add up

Reply:
it's no big deal... costs us a few bytes... but maybe better with the more descriptive name